regulation-worker/internal/delete/batch/batch.go

```diff
@@ -144,115 +113,106 @@ func (*Batch) updateStatusTrackerFile(absStatusTrackerFileName, fileName string)
 
-	gzipReader, err := gzip.NewReader(compressedFilePtr)
-	if err != nil {
-		return "", fmt.Errorf("error while reading compressed file: %w", err)
-	}
+	var response []string
+	defer f.Close()
 
-	decompressedFilePtr, err := os.CreateTemp(b.TmpDirPath, "")
+	byt, err := io.ReadAll(f)
 	if err != nil {
-		return "", fmt.Errorf("error while creating temporary file for decompressed files during cleaning")
+		return nil, fmt.Errorf("unable to read contents of status tracker file, err: %w", err)
 	}
-	defer decompressedFilePtr.Close()
+	// if statusTracker.txt exists then read it & remove all those files name from above gzFilesObjects,
+	// since those files are already cleaned.
 
-	_, err = io.Copy(decompressedFilePtr, gzipReader)
-	if err != nil {
-		return "", fmt.Errorf("error while writing uncompressed file: %w", err)
-	}
+	jobID := fmt.Sprintf("%d", job.ID)
 
-	decompressedFileName, err := filepath.Abs(decompressedFilePtr.Name())
-	if err != nil {
-		return "", err
-	}
+	if len(byt) == 0 {
+		// insert <jobID> in 1st line
+		if _, err := io.WriteString(f, jobID+"\n"); err != nil {
+			return nil, fmt.Errorf("error while writing to jobId to statusTrackerFile: %w", err)
+		}
+	} else {
+		lines := strings.Split(string(byt), "\n")
+		// check if our <jobID> matches with the one in file.
+		// if not, then truncate the file & write new current jobID.
 
-	return decompressedFileName, nil
-}
+		// TODO: Why would this happen ? ( remanent of the past )
+		if lines[0] != jobID {
+			f.Close()
 
-// compress & write `cleanedBytes` of type []byte to `fileName`
-func (*Batch) compress(fileName string, cleanedBytes []byte) error {
-	// compressing
-	var buffer bytes.Buffer
-	w := gzip.NewWriter(&buffer)
-	_, err := w.Write(cleanedBytes)
-	if err != nil {
-		return fmt.Errorf("error while compressing file: %w", err)
-	}
-	w.Close() // must close this first to flush the bytes to the buffer.
+			statusTrackerTmpDir, err := os.MkdirTemp(b.TmpDirPath, "")
+			if err != nil {
+				return nil, fmt.Errorf("error while creating temporary directory: %w", err)
+			}
 
-	cleanCompressedFilePtr, err := os.OpenFile(fileName, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
-	if err != nil {
-		return fmt.Errorf("error while opening file, %w", err)
-	}
-	defer cleanCompressedFilePtr.Close()
+			f, err = os.OpenFile(filepath.Join(statusTrackerTmpDir, StatusTrackerFileName), os.O_CREATE|os.O_RDWR, 0o644)
+			if err != nil {
+				return nil, fmt.Errorf("error while opening file to create status tracker, %w", err)
+			}
 
-	_, err = cleanCompressedFilePtr.Write(buffer.Bytes())
-	if err != nil {
-		return fmt.Errorf("error while writing cleaned & compressed data:%w", err)
+			if _, err := io.WriteString(f, jobID+"\n"); err != nil {
+				return nil, fmt.Errorf("error while writing to status tracker file:%s, err: %w", StatusTrackerFileName, err)
+			}
+
+		} else {
+			// if yes, then simply read it.
+			response = lines[1 : len(lines)-1]
```

Comment:
This assumes that the file contains at least one line. We should valid the file before making the assumption, or use a common format for storing this information. 

(not introduced in this PR)

Reply:
Good point, I'll come up with a plan for fixing it. Potential nil pointer exception.